eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -418,6 +422,17 @@ public synchronized BackupRegistry get() {
             fetchRegistryFromBackup();
         }
 
+        // Configure new transport layer (candidate for injecting in the future)
+        EurekaHttpClient newEurekaHttpClient = null;
+        try {
+            String[] availZones = clientConfig.getAvailabilityZones(clientConfig.getRegion());
+            String myZone = InstanceInfo.getZone(availZones, instanceInfo);
+            newEurekaHttpClient = EurekaHttpClients.createStandardClient(clientConfig, applicationInfoManager, myZone);
```

Comment:
Can probably move the myZone initialization into createStandardClient, since it already takes clientConfig and appInfoManager as params.